### ISSUE TITLE
Related links in PDF should not show child links by default

### ISSUE THREAD

**Issue by raducoravu** [dita-ot/dita-ot#1887]
As an example:

My DITA Map does not specify the collection-type attribute anywhere.
I have some topicrefs inside it:

     <topicref href="concepts/autumnFlowers.dita" >
      <topicref href="topics/flowers/chrysanthemum.dita"/>
    </topicref>

I generate the PDF specifying "args.rellinks=all".
The PDF output contains for the "autumnFlowers.dita" topic also related links to its child topics.
The XHTML output does not contain the related link to the child.

In my opinion to make them consistent the XSLT stylesheet:

     DITA-OT\plugins\org.dita.pdf2\xsl\fo\links.xsl

has an XSLT template:

      <xsl:template match="*[contains(@class,' topic/link ')]">

in which before the &lt;xsl:otherwise&gt; clause we could inhibit generating child links if there is no parent with collection-type="sequence":

     <xsl:when test="@role='child' and not(parent::*/@collection-type='sequence')"/>

**Comment by raducoravu** [comment]
Ideally we would also disable parent links in the PDF output by adding an extra:

      <xsl:when test="@role='parent'"/>

because they are not useful for anything.

**Comment by jelovirt** [comment]
Internally, `args.rellinks` defines default values for [`include.rellinks`](https://github.com/dita-ot/dita-ot/blob/develop/src/main/plugins/org.dita.base/build_init.xml#L257). This can be set to exactly define which roles are output.

**Comment by raducoravu** [comment]
Good idea, that is indeed a very good way to better control links in the PDF output.
Maybe still for the child links we could add consistency between XHTML and PDF outputs.

**Comment by robander** [comment]
For what it's worth - I usually do not want those two to be consistent - when I'm generating XHTML I always want child links (they usually go to a different XHTML page), but in PDF I never want them.

**Comment by jelovirt** [comment]
I agree with @robander, different media might/do have different requirements. I would prefer to keep child/parent links out from paginated/single document media.

**Comment by raducoravu** [comment]
When the XHTML output generates child links, it also looks at collection-type attributes before doing so. The PDF output generates these by default. So I think this is inconsistency which could be fixed.

Indeed when people enable linking in the PDF output they probably do not want child, parent, next or previous links to appear at all. They probably want the explicit links defined in the relationship table to appear.

**Comment by bridgetr** [comment]
I am having exactly this issue. All links are carried into the PDF including the Previous Topic and Next Topic links along with arrows. Is there a way to switch this off for PDF generation (without having to remove the collection-type from the map)?
Thanks

**Comment by bridgetr** [comment]
By the way, I do have the args.rellinks set to nofamily. And in my map I'm using the collection-type="sequence" and not using the linking param.

**Comment by jelovirt** [comment]
@bridgetr please use either DITA Users or DITA-OT Users mailing list for support.
